pkg/network/setup/netpod/discoverbridge.go

```diff
@@ -121,14 +121,21 @@ func translateNmstateToNetlinkRoutes(otherRoutes []nmstate.Route) ([]vishnetlink
 // filterRoutesByNonLocalDestination filters out local routes (the destination is of the local link).
 // Default routes should not be filter out.
 func filterRoutesByNonLocalDestination(linkRoutes []nmstate.Route, addr *vishnetlink.Addr) ([]nmstate.Route, error) {
+	_, localNetwork, err := net.ParseCIDR(addr.String())
+	if err != nil {
+		return nil, err
+	}
+
+	localNetworkStr := localNetwork.String()
+
 	var otherRoutes []nmstate.Route
 	for _, route := range linkRoutes {
 		_, dstIPNet, perr := net.ParseCIDR(route.Destination)
 		if perr != nil {
 			return nil, perr
 		}
 		isDefaultRoute := route.Destination == nmstate.DefaultDestinationRoute(vishnetlink.FAMILY_V4).String()
-		localDestination := !isDefaultRoute && dstIPNet.Contains(addr.IP)
+		localDestination := !isDefaultRoute && dstIPNet.String() == localNetworkStr
```

Comment:
instead using `dstIPNet.String()` you can keep localNetwork as int and just compare both please